azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -0,0 +1,979 @@
+	}
+
+	if v, ok := d.GetOk("availability_set_id"); ok {
+		params.AvailabilitySet = &compute.SubResource{
+			ID: utils.String(v.(string)),
+		}
+	}
+
+	if v, ok := d.GetOk("custom_data"); ok {
+		params.OsProfile.CustomData = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("dedicated_host_id"); ok {
+		params.Host = &compute.SubResource{
+			ID: utils.String(v.(string)),
+		}
+	}
+
+	if evictionPolicyRaw, ok := d.GetOk("eviction_policy"); ok {
+		if params.Priority != compute.Spot {
+			return fmt.Errorf("An `eviction_policy` can only be specified when `priority` is set to `Spot`")
+		}
+
+		params.EvictionPolicy = compute.VirtualMachineEvictionPolicyTypes(evictionPolicyRaw.(string))
+	} else if priority == compute.Spot {
+		return fmt.Errorf("An `eviction_policy` must be specified when `priority` is set to `Spot`")
+	}
+
+	if v, ok := d.Get("max_bid_price").(float64); ok && v > 0 {
+		if priority != compute.Spot {
+			return fmt.Errorf("`max_bid_price` can only be configured when `priority` is set to `Spot`")
+		}
+
+		params.BillingProfile = &compute.BillingProfile{
+			MaxPrice: utils.Float(v),
+		}
+	}
+
+	if v, ok := d.GetOk("proximity_placement_group_id"); ok {
+		params.ProximityPlacementGroup = &compute.SubResource{
+			ID: utils.String(v.(string)),
+		}
+	}
+
+	if v, ok := d.GetOk("zone"); ok {
+		params.Zones = &[]string{
+			v.(string),
+		}
+	}
+
+	// "Authentication using either SSH or by user name and password must be enabled in Linux profile." Target="linuxConfiguration"
+	adminPassword := d.Get("admin_password").(string)
+	if disablePasswordAuthentication && len(sshKeys) == 0 {
+		return fmt.Errorf("At least one `admin_ssh_key` must be specified when `disable_password_authentication` is set to `true`")
+	} else if !disablePasswordAuthentication {
+		if adminPassword == "" {
+			return fmt.Errorf("An `admin_password` must be specified if `disable_password_authentication` is set to `false`")
+		}
+
+		params.OsProfile.AdminPassword = utils.String(adminPassword)
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, params)
+	if err != nil {
+		return fmt.Errorf("Error creating Linux Virtual Machine %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Linux Virtual Machine %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving Linux Virtual Machine %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Error retrieving Linux Virtual Machine %q (Resource Group %q): `id` was nil", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+	return resourceLinuxVirtualMachineRead(d, meta)
+}
+
+func resourceLinuxVirtualMachineRead(d *schema.ResourceData, meta interface{}) error {
```

Comment:
there's a mix across resources - tbh I'm not sure I see the value of sorting these methods, since they're cmd+clickable from the resources block anyway?